zoneminder: Don't attempt to use alarm status API on older versions.

tobygray opened the pull request:
## Description:
The alarm status API was introduced in Zoneminder 1.30. This commit
prevents any attempt to retreive the alarm status when connected to
Zoneminder servers older than 1.30.

This prevents the log from filling up with "Could not get status
for monitor 1" messages.

**Related issue (if applicable):** fixes #7679

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** No documentation update necessary.

## Example entry for `configuration.yaml` (if applicable):

No configuration change needed.

## Checklist:

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

tobygray committed:
zoneminder: Don't attempt to use alarm status API on older versions.

The alarm status API was introduced in Zoneminder 1.30. This commit
prevents any attempt to retreive the alarm status when connected to
Zoneminder servers older than 1.30.

This prevents the log from filling up with "Could not get status
for monitor 1" messages.

tobygray committed:
Fix style violations

tobygray committed:
Fix flake8 documentation complaints

tobygray commented on homeassistant/components/zoneminder.py:
```diff
@@ -40,6 +41,27 @@
+# Introduced in 1.30:
+#
+# https://github.com/ZoneMinder/ZoneMinder/commit/2888142e682bbc9950535d7e5aaef2cd20cda38d
+ALARM_STATUS_FEATURE = _Feature("1.30")
```
Fixed. I now just need to remember that flake8 is used as one of the linters and not pyflakes...

balloob commented on homeassistant/components/zoneminder.py:
```diff
@@ -7,6 +7,7 @@
 import logging
```
Please revert the file permission change

balloob replied:
This still needs addressing.

tobygray replied:
I thought I had done. Is 644 not the correct permissions?

tobygray committed:
Fix accidental file permissions change

balloob commented on homeassistant/components/zoneminder.py:
```diff
@@ -40,6 +41,29 @@
+        Returns False if this feature isn't supported on the currently
+        connected zoneminder instance. Otherwise returns True.
+        """
+        remote_version = ZM.get('version', None)
```
We should move away from using globals. It's out of scope for this PR to fix that but please don't introduce new usage of the globals but instead have the `ZM` object be passed in.

Actually, it's kinda overkill to introduce a whole class for this. Why not just store `alarm_status_supported` in the ZM dict in the setup?

tobygray replied:
I wanted to avoid adding a reference to `zoneminder.ZM` in the `camera.zoneminder` module and thought that adding a class would make it easier to support additional features. How about I add a `zoneminder.alarm_status_supported()` function for `camera.zoneminder` to call?

I'm happy to do the refactoring to get rid of the ZM object. Is there a different module that I can use as a good reference? Would move `zoneminder.ZM` to `hass.data['zoneminder']` like `nest` be the thing to do?

balloob replied:
Yes, `hass.data` is the right approach. 

I also like `zoneminder.alarm_status_supported()` idea !

tobygray committed:
Merge remote-tracking branch 'upstream/dev' into zoneminder-conditional-support

tobygray committed:
Undo permission changes.

This reverts the file permission changes introduced in:
* https://github.com/home-assistant/home-assistant/pull/7673
* https://github.com/home-assistant/home-assistant/pull/7675

tobygray committed:
zoneminder: Move global ZM dictionary into hass.data.

As suggested by @balloob this commit removes the use of the
`zoneminder.ZM` global dict and replaces it with
`hass.data['zoneminder']`.

tobygray committed:
zoneminder: Fix linter issues

mnoorenberghe reviewed (changes requested):
> This commit
> prevents any attempt to retreive the alarm status when connected to
> Zoneminder servers older than 1.30.

ZoneMinder older than 1.30.x has major security issues and is many years old so I don't think it should be supported by Home Assistant, especially if it adds as much complexity as this PR does. I think it would be acceptable for users who don't want to upgrade ZM to simply lower the log level for the zoneminder component.

pvizeli commented:
I agree with @mnoorenberghe

tobygray commented:
It sounds like I should just close this PR and wait for Debian "stretch" to be released with a newer version of Zoneminder.

pvizeli commented:
Can you run zoneminder as docker? So I can make hass.io addon for it

tobygray commented:
There are a few docker images for zoneminder around, but I've never tried any of them so I don't know how well maintained they are.

balloob commented:
Alright, closing this PR then. Stretch is around the corner anyway ([5 days!](https://lists.debian.org/debian-devel-announce/2017/05/msg00002.html))
